applications/openshift/api-server/api_server_oauth_https_serving_cert/rule.yml

```
@@ -0,0 +1,50 @@
+documentation_complete: true
+
+prodtype: ocp4
+
+title: 'Make sure that the openshift-oauth-apiserver service uses TLS'
+
+description: |-
+    By default, the OpenShift OAuth API Server uses TLS. HTTPS should be
+    used for connections between openshift-oauth-apiserver and kube-apiserver.
+    OpenShift OAuth API server enables TLS automatically if a TLS key and a
+    certificate are provided via the <tt>serving-cert</tt> secret
+    in the <tt>openshift-oauth-apiserver</tt> namespace.
+
+rationale: |-
+    Connections between the kube-apiserver and the extension
+    openshift-oauth-apiserver could potentially carry sensitive data such
+    as secrets and keys. It is important to use in-transit encryption
+    for any communication between the kube-apiserver and the extension
+    openshift-apiserver.
+
+severity: medium
+
+references:
+    cis: 1.2.4
+
+ocil_clause: |-
+    Verify that the serving-cert for the openshift-oauth-apiserver is type
+    <tt>kubernetes.io/tls</tt> and that returned Data includes <tt>tls.crt</tt>
+    and <tt>tls.key</tt>.
```

The `ocil_clause` when fully built looks like:

```
Is it the case that Verify that the serving-cert for the openshift-apiserver
is type <tt>kubernetes.io/tls</tt> and that returned Data includes
<tt>tls.crt</tt> and <tt>tls.key</tt>.?
```

which is a little weird, so would recommend something like:

```suggestion
    the openshift-apiserver serving-cert is not set to type
    <tt>kubernetes.io/tls</tt> and that returned Data does not include <tt>tls.crt</tt>
    and <tt>tls.key</tt>
```

---

fixed, thanks